### DISCUSSION TITLE
Can we add localhost mysql database inside budibase?

### DISCUSSION THREAD

**Discussion by AryanGaur14** [Budibase/budibase#13388] [Help] [answered]
Error creating datasource: Unable to connect: connect ECONNREFUSED 127.0.0.1:3306 this error pops up when i try to add.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @AryanGaur14,

Yes you should be able to add your own local databases to Budibase. This will likely only work if you are self-hosting Budibase. If you are attempting this whilst using Cloud you won't be able to connect.

If you are self-hosted have you tried using `host.docker.internal` as the host setting?

**Reply by Townieeee** [reply]
Appreciate your reply. I saw your original question from 3 weeks ago. I am new to bb and the hosting too - so find even the terminology difficult. I thought I could use bb in the cloud and connect to MYSQL on my local laptop as a testing ground. I however just learnt this is not possible as MYSQL needs to be on some cloud service and even then it has to be setup to accept requests coming in from the web. You have suggested I install bb on my laptop (I think that is what you mean when you say This will likely only work if you are self-hosting Budibase ). This is a great suggestion - thx - I am so new to bb that I thought it was "only" cloud based. Looks like I have more digging to try. Again -appreciate yr help!!

**Reply by Townieeee** [reply]
Please tell me what am I missing: I signed up for 3 day trial on Cloudways that has MYSQL for me to test on. I pasted a Mysql script to create my MYSQL test DB with a few records. All was looking good. I had my parameters ready (host,port,user,password and database ) but get (Guessed the 3306 for port) "Error creating datasource: Unable to connect: connect ETIMEDOUT". This is the part where I just don't know what to do as I thought all I needed was the parameters. There is 1 more parameter that Budibase has on the CONNECT TO MYSQL AND THAT IS SSL. Do I need that too to test if my data can be used by BB? Hopefully not. I got BB to work recently by connecting to a gsheet, but need to do same with Mysql. There must be some fellow buddies here who have also go this far but just cannot get over the finish line. Any help is appreciated. thx
